src/Payum/Be2Bill/Api.php

```diff
@@ -120,10 +124,11 @@ class Api
     /**
      * @param array               $options
      * @param HttpClientInterface $client
+     * @param MessageFactory      $messageFactory
      *
      * @throws \Payum\Core\Exception\InvalidArgumentException if an option is invalid
      */
-    public function __construct(array $options, HttpClientInterface $client = null)
+    public function __construct(array $options, HttpClientInterface $client, MessageFactory $messageFactory )
```

Comment:
Before I rewrite the failing tests. Is this the function signature that yo want?